File: examples/projects/benchmarks/nqueens/nqueens.swift
// 基准: nqueens —— 12 皇后回溯搜索（纯整数 + 递归 + 分支）
// 与 nqueens.zeta 逻辑严格一致。输出 = 14200
func place(_ queens: inout [Int64], _ row: Int64, _ n: Int64) -> Int64 {
    if row == n { return 1 }
    var total: Int64 = 0
    for col in 0..<n {
        var ok = true
        for r in 0..<row {
            let q = queens[Int(r)]
            if q == col || q - r == col - row || q + r == col + row {
                ok = false
                break
            }
        }
        if ok {
            queens.append(col)
            total += place(&queens, row + 1, n)
            queens.removeLast()
        }
    }
    return total
}

var queens = [Int64]()
print(place(&queens, 0, 12))

File: examples/projects/benchmarks/nqueens/nqueens.go
// 基准: nqueens —— 12 皇后回溯搜索（纯整数 + 递归 + 分支）
// 与 nqueens.rl 逻辑严格一致。输出 = 14200
package main

import "fmt"

func place(queens []int64, row int64, n int64) int64 {
	if row == n {
		return 1
	}
	var total int64
	for col := int64(0); col < n; col++ {
		ok := true
		for r := int64(0); r < row; r++ {
			q := queens[r]
			if q == col || q-r == col-row || q+r == col+row {
				ok = false
				break
			}
		}
		if ok {
			queens[row] = col
			total += place(queens, row+1, n)
		}
	}
	return total
}

func main() {
	queens := make([]int64, 12)
	fmt.Println(place(queens, 0, 12))
}
